### ISSUE TITLE
NET6 Mobile projects do not load in VS after updating Uno templates and Uno.Check crashes

### ISSUE THREAD

**Issue by mikernet** [unoplatform/uno#7596]
### Current behavior

After updating Uno templates to the latest non-dev version, running `dotnet new unoapp-net6` (which was working before updating the templates) started producing errors that said something along the lines of `cannot build this project, workload macos missing. Run 'dotnet workload install macos'  to fix the issue` (that wording is not exact, I can't recall 100% what it said now), so I did that and repeated for each workload it complained about (which I believe was `android, ios, maccatalyst` and `macos`) until `dotnet new unoapp-net6` worked.

Now the `.Mobile` projects created by the template and the ones I had created before updating the templates no longer load in VS - it says project is incompatible:

![image](https://user-images.githubusercontent.com/1700111/144739419-a0cace71-89e7-4815-8e01-ab83652fbd22.png)

I tried various combinations of the target frameworks by editing the csproj and it seems like none of them work. The NET6 WASM project loads and builds. My old non-net6 projects still load and build. 

The `.Mobile` project builds from the command line.

I tried updating `uno-check` to the latest non-dev version and running that, and it crashes with:

```
System.InvalidOperationException: Could not find color or style 'Argument'.
  at Spectre.Console.StyleParser.Parse(String text) in /_/src/Spectre.Console/StyleParser.cs:13
  at Spectre.Console.MarkupParser.Parse(String text, Style style) in /_/src/Spectre.Console/Internal/Text/Markup/MarkupParser.cs:26
  at Spectre.Console.AnsiConsoleExtensions.Markup(IAnsiConsole console, String value) in /_/src/Spectre.Console/Extensions/AnsiConsoleExtensions.Markup.cs:41
  at Spectre.Console.AnsiConsoleExtensions.MarkupLine(IAnsiConsole console, String value) in /_/src/Spectre.Console/Extensions/AnsiConsoleExtensions.Markup.cs:62
  at Spectre.Console.AnsiConsole.MarkupLine(String value) in /_/src/Spectre.Console/AnsiConsole.Markup.cs:46
  at DotNetCheck.Cli.CheckCommand.checkupStatusUpdated(Object sender, CheckupStatusEventArgs e) in /_/UnoCheck/CheckCommand.cs:319
  at DotNetCheck.Checkups.WSLCheckup.Examine(SharedState history) in /_/UnoCheck/Checkups/WSLCheckup.cs:34
  at DotNetCheck.Cli.CheckCommand.ExecuteAsync(CommandContext context, CheckSettings settings) in /_/UnoCheck/CheckCommand.cs:167

> .NET SDK - Workloads (6.0.100) Checkup...
  - android-aot (Microsoft.NET.Sdk.Android.Manifest-6.0.100 : 31.0.101-preview.11.81) installed.
  - ios (Microsoft.NET.Sdk.iOS.Manifest-6.0.100 : 15.0.101-preview.11.391) installed.
  - maccatalyst (Microsoft.NET.Sdk.MacCatalyst.Manifest-6.0.100 : 15.0.101-preview.11.391) installed.
  - tvos (Microsoft.NET.Sdk.tvOS.Manifest-6.0.100 : 15.0.101-preview.11.391) installed.
  - macos (Microsoft.NET.Sdk.macOS.Manifest-6.0.100 : 12.0.101-preview.11.391) installed.
  - maui (Microsoft.NET.Sdk.Maui.Manifest-6.0.100 : 6.0.101-preview.10.2068) installed.
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

Checkup had Error status: wsl

wsl: Could not find color or style 'Argument'.
! There were one or more problems detected.
Please review the errors and correct them and run uno-check again.
```

Not really sure how to proceed from here.

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia

### Affected platforms

iOS, Android, macOS

**Comment by jeromelaban** [comment]
Thank for the report.

For net6 mobile, make sure to use VS 2022 17.1 Preview 1 or later, earlier versions don't support it.

For uno-check, you're likely hitting https://github.com/unoplatform/uno.check/issues/11, which you can safely skip (or run with `--skip wsl`).

**Comment by mikernet** [comment]
Oh, I see, that makes sense I guess. Not sure why it was working before, must have been using an older Xamarin verison?

Thank you, will give that a try with the preview shortly.

**Comment by marwalsch** [comment]
@jeromelaban Is this restriction planned to be removed or do have to wait for a stable 17.1 version? The VS22 preview is incredibly quirky with everything related to Xaml, yet targeting .NET 5 instead of 6 for new projects at the times makes little sense by now.

**Comment by jeromelaban** [comment]
@marwalsch this is not a restriction that we manage, but rather the .NET team. .NET 6 mobile support is only available in 17.1 preview 1 and later, and that will not change.
